src/OpenFeature.DependencyInjection/Internal/FeatureLifecycleManager.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace OpenFeature.Internal;
+
+internal sealed class FeatureLifecycleManager : IFeatureLifecycleManager
+{
+    private readonly Api _featureApi;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<FeatureLifecycleManager> _logger;
+
+    public FeatureLifecycleManager(Api featureApi, IServiceProvider serviceProvider, ILogger<FeatureLifecycleManager> logger)
+    {
+        _featureApi = featureApi;
+        _serviceProvider = serviceProvider;
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public async ValueTask EnsureInitializedAsync(CancellationToken cancellationToken = default)
+    {
+        _logger.LogInformation("Starting initialization of the feature provider");
+        var featureProvider = _serviceProvider.GetService<FeatureProvider>();
+        if (featureProvider == null)
+        {
+            throw new InvalidOperationException("Feature provider is not registered in the service collection.");
+        }
+        await _featureApi.SetProviderAsync(featureProvider).ConfigureAwait(false);
```

Comment:
We might want an optional value for domain, or some other method for setting [domain-scoped providers](https://github.com/open-feature/dotnet-sdk/blob/c471c062cf70d78b67f597f468c62dbfbf0674d2/README.md?plain=1#L154).